src/inspect.rs

```diff
@@ -124,7 +120,8 @@ impl MetadataPrinter {
                     self.print_metadata_context_aware_resources(metadata, no_color)?;
                     println!();
                 }
-                self.print_metadata_usage(metadata, no_color)
+                self.print_metadata_usage(metadata, no_color);
+                Ok(())
```

Comment:
You can remove this line and remove the one on 112. Then change the signature of this method to not be just void. We don't need to have `Result<()>` anymore, since the new code is infallible

Reply:
for some of those `print_metadata` we still serialize with serde, or check the protocol_version and error, so I think we can't do that.